src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -132,10 +144,17 @@
     public MonitoredTrip() {
     }
 
-    public MonitoredTrip(OtpDispatcherResponse otpDispatcherResponse) {
+    public MonitoredTrip(OtpDispatcherResponse otpDispatcherResponse) throws URISyntaxException {
         queryParams = otpDispatcherResponse.requestUri.getQuery();
         TripPlan plan = otpDispatcherResponse.getResponse().plan;
         itinerary = plan.itineraries.get(0);
+
+        // extract trip time from parsed params
+        Map<String, String> params = parseQueryParams();
+        tripTime = params.get("time");
+        if (tripTime == null) {
```

Comment:
To follow up on https://github.com/ibi-group/otp-middleware/pull/75#discussion_r489636219,
the UI does not currently initialize `tripTime`, so `tripTime` also needs to be initialized when polling `tripTimeHour` or `tripTimeMinute`. (Those methods are called by `CheckMonitoredTrip#shouldSkipMonitoredTripCheck`).

Reply:
Ah, ok I see what's happening. I added some logic to the MonitoredTripController's preCreateHook and preUpdateHook to enforce the existence of the trip time before it gets saved to the database.